airflow/www/views.py

```diff
@@ -3380,12 +3381,26 @@ def action_muldelete(self, items):
     def pre_add(self, item):
         """Pre add hook."""
         item.execution_date = timezone.make_aware(item.execution_date)
-        item.value = XCom.serialize_value(item.value)
+        item.value = serialize_value_shim(
+            value=item.value,
+            key=item.key,
+            task_id=item.task_id,
+            dag_id=item.dag_id,
+            execution_date=item.execution_date,
+            run_id=item.run_id,
+        )
 
     def pre_update(self, item):
         """Pre update hook."""
         item.execution_date = timezone.make_aware(item.execution_date)
-        item.value = XCom.serialize_value(item.value)
+        item.value = serialize_value_shim(
```

Comment:
I'm not a fan of this from an API -- we should instead _shim_/wrap the old backends in resolve backend, something like

```python
    serialize_value = clazz.serialized_value
    if old_serialize(serialized_value):
        def shim(value: **kwargs):
            return serialize_value(value)
        clazz.serialize_value = shim
```

Reply:
sure lemme try that